apps/fluent-tester/src/E2E/RadioGroup/pages/RadioGroupPageObject.win.ts

```diff
@@ -58,12 +104,22 @@ class RadioGroupPage extends BasePage {
   /***************/
   /* RadioButton *
   /**************/
-  get _radioButton() {
-    return By(RADIOBUTTON_TEST_COMPONENT);
+
+  /* The first RadioGroup has 4 radio buttons, all listed below */
+  get _firstRadioButton() {
```

Comment:
I think the test pages would be more flexible if you could have a 

```
get _radioButtonById(id: string) {
    return By(id);
}
```

sort of thing instead. Then you wouldn't need to add a get function for each component you want to track.

It also seems to me that we could use the testIds to refer to components instead of the Selectors, is there a reason to use the Selectors over the testId strings? I can think of type safety being a reason, but otherwise it seems like it's making the process of referring to elements more complex.

Reply:
I definitely see where you're coming from. The answer is: We don't want the tests themselves to be using the Test IDs. We're using the [Page Object Pattern](https://webdriver.io/docs/pageobjects) (recommended by WebDriverIO).

The goal of using Page Objects is to abstract any page information away from the actual tests. Ideally, you should store all selectors or specific instructions that are unique for a certain page in a page object, so that you still can run your test after you've completely redesigned your page.

I do see what you're saying about making it easier to track, though. That's why I added `getRadioButton(radioButtonSelector: RadioButtonSelector)` that makes it easier for the spec to communicate with the Page Object.

Another reason why we wouldn't necessarily want something like  
```
get _radioButtonById(id: string) {
    return By(id);
}
```
is because this will query the driver every single time we want to interact with the app, rather than just one time when the Page Object class gets initialized. It would effect the perf and opens the door for more intermittent failures (the more unnecessary times we query the driver, the higher chance a failure might happen.

But I do see what you're saying and I'd be open to having a discussion about pros/cons for each method.


Additionally, for using the TestIDs to refer to components instead of selectors. I'm a bit confused by the question, but I'll try to explain the functionality of both... Let me know if I missed the question.

The "Selectors" are the methods of querying the app driver for specific UI components. The selector method needs some type of unique id to query these elements.. Hence the TestID. We need the "Selectors" to query into the app to return the specific UI element. We need the TestID to specify which UI element we want.

Reply:
I see, didn't realize that this was a suggested pattern and that there are perf reasons behind this. Thanks for the explanation!